src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -32,7 +32,7 @@
 /**
  * <p>
  * Checks the placement of right curly braces ({@code '}'}) for code blocks. This check supports
- * if-else, try-catch-finally blocks, while-loops, for-loops,
+ * if-else, try-catch-finally blocks, switch statements and expressions, while-loops, for-loops,
```

Comment:
@rnveach please look at https://github.com/checkstyle/checkstyle/pull/12640#discussion_r1070518408 we have this type of different test cases in the code. which contains switch the expression as well. this is not limited to switch statement

Reply:
All the type of test cases listed say switch in some form or another. They do not sure pure expressions which is what I infer from `and expressions`. The wording needs to be more specific as we don't validate general expressions.

Reply:
See https://github.com/checkstyle/checkstyle/pull/12640#discussion_r1111132009

Expressions should not be evaluated.

Reply:
@rnveach should i remove entirely all the expression or need to remove https://github.com/checkstyle/checkstyle/pull/12640#discussion_r1111132009 only

Reply:
Documentation should not mention "expressions" generally. You need to mention specific ones which should only be switch statements. We should not be validating switch expressions as part of this issue. So documentation should say "and switch statements".

Reply:
@nrmancuso See https://github.com/checkstyle/checkstyle/pull/12640#discussion_r1111132551 and the issue it points to.

Reply:
done

Reply:
@Kevin222004 Please confirm, you removed all expression support you added for `RightCurly`? Tests will show it prints no violations on those except the original `switch () {}`.

Reply:
@rnveach Please once look at the test case which are created at ResourceNonComplilable test case i have removed all the extra things added like https://github.com/checkstyle/checkstyle/pull/12640#discussion_r1111116727 --> I have not removed this test cases but after updating it will be ok know 

condition:- 1
it will support to test like 
```
    public void test5() {
        int num = 0;
        switch (num) {
            case 1 -> System.out.println("One");
            default -> System.out.println("Other");
        } // ok
    }
```
should i have to remove this as well ?

only switch case like should be supported 
```
    public static void method0() {
        int mode = 0;
        switch (mode) {
            case 1:
                int x = 1;
                break;
            default :
                x = 0; } // violation '}' at column 24 should be alone on a line'
    }
```

please let me know should i remove completely all the expression related to condion 1

Reply:
I Confirmed that all the expression has removed but i cant remove the test cases added for expression where ever their is violation it is ok know, because the test cases has taken from previous generated regression

done

Reply:
```
            c = switch (a) {
                case 0 -> 5;
                case 1 -> 10;
                default -> switch (b) {
                    case 2 -> {
                        if (a == 0) {
                            d = 1; // reassign d
                        }
                        throw new Exception();
                    }
                    default -> 2; }; // ok  --> showing violation
            }; // ok
        }
```
This case is still pending to solve for further review i have make it ok but this case is becoming complex :)

Reply:
I don't understand.

This check should not validate expressions. I am completely fine with test cases staying that show they will NOT be validated now, but there can't be any validation or violations. This check doesn't support expressions and it would take @romani 's approval to say it does now.

Reply:
https://github.com/checkstyle/checkstyle/pull/12640#discussion_r1127545991 This i not supported as it is an expression 

Done this type of test cases we have in tests and it will be skipped as it is an expression